After download callback for artifact_package

aaronfeng opened the pull request:
Call the after_download proc if an artifact has been downloaded, otherwise it's not called.

aaronfeng committed:
call back if the package has been downloaded

KAllan357 commented:
Do you think this would be helpful if it was exposed to the `artifact_deploy` resource as well?

I think you'd just want to pass it through on the `artifact_file` resources linked here - https://github.com/RiotGames/artifact-cookbook/blob/master/providers/deploy.rb#L542-L599

Can this have tests added? I think the test could be added in `file.rb` that can be run and verified by test-kitchen.

aaronfeng committed:
default after_download proc to do nothing

aaronfeng committed:
add integration test for after_download callback

aaronfeng committed:
nexus classifier bug

the original code assumed whatever value after
the extension is the version.  This is not
necessary true since the classifier can come
between the version and extension.

aaronfeng committed:
integrated after_download to deploy resource

- included integration tests

KAllan357 commented on libraries/chef_helpers.rb:
```diff
@@ -0,0 +1,26 @@
+      # @param name     [Symbol] the name of the proc to execute
+      #
+      # @return [void]
+      def execute_run_proc(from, resource, name)
```
Is it possible to remove the `from` parameter and use `#{__FILE__}` to get us that extra bit of logging?

It might make it cleaner to have this method be called `run_proc` and then remove the private methods on the other classes.

aaronfeng replied:
I don't think you can if you want to preserve the same style of logging before.

KAllan357 commented on libraries/chef_helpers.rb:
```diff
@@ -0,0 +1,26 @@
+      # numerous Proc attributes defined for this resource.
+      #
+      # @param from     [String] location where the proc is executed from
+      # @param resource [Chef::Resource::ArtifactDeploy] resource being executed on
```
Maybe just `Chef::Resource` here

aaronfeng committed:
wire after_download hook to deploy resource

aaronfeng committed:
include nexus package test

aaronfeng committed:
Merge remote-tracking branch 'celkins/feature/support-maven-classifier' into after_download

    Conflicts:
        providers/deploy.rb

aaronfeng committed:
include nexus_package as default integration test

aaronfeng committed:
include after_download in readme

KAllan357 commented:
:+1:
